### PR TITLE
Enforce Recidiviz copyright header

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

Enforce Recidiviz copyright header using Eslint plugin

Screen Shot 2022-10-04 at 12 07 51 PM

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  extends: ["@recidiviz"],
+  plugins: ["header"],
+  rules: {
+    "header/header": [2, "license-header.js"],
```
Comment (mxosman) [inline publisher/.eslintrc.js:5]: Do you now what the `2` or that position in the array represents?

**Comment by terryttsai** [comment]
So in 2023, if we want to update every header to say 2023, we should be able to update the template and run autofix for it to do that for us.

If we only want new files moving forward to say 2023, and allow old files to say 2022, then we could configure it using a regular expression to allow the year to be 2022, 2023, 2024 etc. See https://github.com/Stuk/eslint-plugin-header#regular-expressions

**Comment by mxosman** [comment]
> So in 2023, if we want to update every header to say 2023, we should be able to update the template and run autofix for it to do that for us.
> 
> If we only want new files moving forward to say 2023, and allow old files to say 2022, then we could configure it using a regular expression to allow the year to be 2022, 2023, 2024 etc. See https://github.com/Stuk/eslint-plugin-header#regular-expressions

What is standard convention for these copyright header years? Every year all files are updated to the current year - or each file has their own "creation" year header?

If we use the regex, would we still have to manually update the file year?

These are just all curiosities! Not sure if it's worth it to go down any route if it feels easy enough for us to remember and update the template annually.

**Comment by terryttsai** [comment]
> What is standard convention for these copyright header years? All files are updated to the current year header - or each file has their own "creation" year header?

Not sure, I heard from Ian that it's not critical but it's not a bad idea to update all files to match the current year

> If we use the regex, would we still have to manually update the file year?

Yeah we'd still have to at least change the template once a year

**Pull request merged**